src/virtualenv/seed/embed/base_embed.py

```diff
@@ -20,21 +20,23 @@ def __init__(self, options) -> None:
 
         self.pip_version = options.pip
         self.setuptools_version = options.setuptools
-        if hasattr(options, "wheel"):
-            # Python 3.8
-            self.wheel_version = options.wheel
-            self.no_wheel = options.no_wheel
-        elif options.no_wheel:
+        self.wheel_version = options.wheel
+
+        self.no_pip = options.no_pip
+        self.no_setuptools = options.no_setuptools
+        self.no_wheel = options.no_wheel
+
+        if options.no_wheel or options.wheel:  # and interpreter.version_info[:2] >= (3, 9)
```

Comment:
This is where I am stuck. I want to check the interpreter version, but this method does not know the interpreter.

The possible but very hacky thing is to add a hidden parser argument `--wheel-is-deprecated` with a default value based on interpreter version and read it here. But that's not nice.

Reply:
Yeah, I hit the same problem when I implemented this. I was rather pleased with the `hasattr` hack, but indeed the whole API needs a refactor to make this work...

Reply:
Yeah we need to pass it through, however maybe this should not be inside the init 😊 but rather inside the run within https://github.com/hroncok/virtualenv/blob/wheel-option-back/src/virtualenv/seed/seeder.py#L31

Reply:
Does the creatoer there hold a reference to interpreter?

Reply:
Or we support all these options here (altering CLI options based on another CLI option (`-p`) is already weird), and make the decision on whether to do something or nothing later in the stack.

Reply:
> Does the creatoer there hold a reference to interpreter?

Yes.